src/services/offer.js

```diff
@@ -65,8 +65,12 @@ class OfferService {
 
     }
 
-    getOfferById(id) {
-        return Offer.findById(id);
+    async getOfferById(req) {
+        const offer = await Offer.findById(req.params.offerId);
+
+        if (offer?.isHidden && !(req.user?.isAdmin || offer.owner.toString() === req.user?.company?._id.toString())) return null;
+
+        return offer;
```

Comment:
The service should be unaware of the routing logic, pass the user and the offerId instead

```suggestion
    async getOfferById(offerId, user) {
        const offer = await Offer.findById(offerId);

        if (offer?.isHidden && !(user?.isAdmin || offer.owner.toString() === user?.company?._id.toString())) return null;

        return offer;
```